### PR TITLE
ENH: Mark control point coordinate items disabled when skipping

### PR DESCRIPTION
**Pull request by jamesobutler**
As discussed at https://discourse.slicer.org/t/new-markups-functionality-retains-unset-control-point-positions/20140/8, marking the control point coordinate items as disabled (grayed out) instead of not displaying the position is a better visual to indicate that a skipped state control point still has a position defined.

I guess control point name could also be set to the disabled look as well if we wanted.

re: #5948

| | Current | This PR|
|-|---------|---------|
|Skip placement | image|image|
|Restore position | image|image|

### CODE REVIEW

**Review by lassoan — COMMENTED** [review]
Thank you for working on this. The changes generally look good to me, but there are a few things to address.

1. Disabling the entire item makes it impossible to select a row by clicking on the coordinate system column. It is quite bad: for about 5 minutes I thought that either my mouse was broken or the widget's internal state was completely messed up. But in the end it turned out that I just cannot click on the coordinate system columns when the items are disabled. Instead of disabling the item we should probably change the text color and set the edit trigger to nothing to prevent editing.

2. I think we should display the grayed out coordinate values even in case of vtkMRMLMarkupsNode::PositionUndefined. I've pushed a commit to this branch that cleans up updating of items and changes this behavior, too (I've been experimenting with this and would have been complicated to separate these two changes, so I just went with committing everything).

For example, if we don't show coordinates in undefined state then the following would happen:
The user defines a point, then clicks "Edit position", hovers over a view, then moves away without clicking then the point state becomes "undefined", therefore coordinates are not showed. In this state the user can click on "Restore position" then some coordinate values appear out of nowhere. Displaying the coordinate values grayed out would make it clear that the last known coordinate values are preserved.

We may consider replacing the point coordinates with some placeholder value (0,0,0) when the user choose to unset the point position to reduce noise/clutter in the RAS columns. (nan, nan, nan) would be more appropriate but then we would need to start checking at dozens of places if the values are valid or nan.

File: Modules/Loadable/Markups/qSlicerMarkupsModuleWidget.cxx [hunk 1]
```diff
@@ -1253,7 +1253,8 @@ void qSlicerMarkupsModuleWidget::updateRow(int m)
   int rColumnIndex = d->columnIndex("R");
   int mPositionStatus = markupsNode->GetNthControlPointPositionStatus(m);
   if (mPositionStatus == vtkMRMLMarkupsNode::PositionDefined ||
-      mPositionStatus == vtkMRMLMarkupsNode::PositionPreview)
+      mPositionStatus == vtkMRMLMarkupsNode::PositionPreview ||
+      mPositionStatus == vtkMRMLMarkupsNode::PositionMissing)
```
[the remark this answers is not in the document]
Comment (jamesobutler) [reply Modules/Loadable/Markups/qSlicerMarkupsModuleWidget.cxx]: @lassoan Thinking about this from a user's perspective if I press "Clear position of highlighted fiducials(s)" and there are still coordinates listed in the table, I'm going to think that is a bug. The usage of "Clear position" makes it appear that the coordinates themselves are going to be cleared to nothing and therefore wouldn't showing anything in the table. An "undefined" control point that has a coordinate position returned makes no logical sense as a simple new user to Slicer.

If there is always going to be a coordinate associated with a control point whether it is Defined, Undefined, Missing, Preview, then some terminology is going to have be improved to make clear status vs position. Although I would be of the opinion for improving the user experience for new users that it makes more logical sense that when you clear or set a control point missing, it should truly have no coordinate position (null position) for it anymore. The current distinction of status vs coordinate position is currently not friendly to new users and is making Markups confusing to use.

image

[the remark this answers is not in the document]
Comment (lassoan) [reply Modules/Loadable/Markups/qSlicerMarkupsModuleWidget.cxx]: I'm coming back to this now. In the past few months we gained some more experience with this, so maybe we can move forward.

What I see is that many modules do not care about control point position status: the module just checks if there is enough number of control points to do something and then it gets the position of the points and uses them (regardless of its status). I see two ways for addressing this:
- A. make `status` weaker: expect that most modules will ignore the status, just use it as some kind of additional hint that certain modules use. We would need to reflect this in the GUI (this PR is a step towards that).
- B. make `status` stronger: only consider a control point as existing if its position is defined. We would need to add new API to get/set properties of undefined control points and hide the undefined control point positions from the GUI.

[the remark this answers is not in the document]
Comment (jamesobutler) [reply Modules/Loadable/Markups/qSlicerMarkupsModuleWidget.cxx]: Yeah seems like most modules won’t use status but a module that uses it specifically for something like Landmarking (primarily Point List node type) would use status.

**Comment by jcfr** [comment]
@jamesobutler Should this one be rebased for integration ?

cc: @smrolfe

**Comment by jcfr** [comment]
Given pending discussions and the fact it is not critical moving to `Backlog`

**Pull request closed without merging**